service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/ReShareCaseController.java

```diff
@@ -48,23 +48,31 @@
     @Secured("caseworker-publiclaw-systemupdate") // NOSONAR
     public void shareCase() {
 
+        int counter = 1;
+
         final String query = new JSONObject()
             .put("from", 0)
             .put("size", 3000)
             .put("query", of("bool", of("must_not",
                 List.of(
                     of("match", of("state", "Deleted")),
                     of("exists", of("field", "supplementary_data"))
-                )
-            )))
+                ),
+                "must", of("match", of("data.caseLocalAuthority", "KBC")))
+            ))
             .toString();
 
+        log.info("Access migration query: {}", query);
+
         final List<CaseDetails> casesDetails = coreCaseDataService.searchCases(CASE_TYPE, query);
 
-        log.info("Number of cases to be migrated: {}", casesDetails.size());
+        log.info("Access migration - number of cases to be migrated: {}", casesDetails.size());
 
         for (CaseDetails casesDetail : casesDetails) {
             try {
+                log.info("Access migration - migrating access for case {}. ({}/{})",
```

Comment:
Could have the loop be
```java
        for (int i = 0, casesDetailsSize = casesDetails.size(); i < casesDetailsSize; i++) {
            CaseDetails casesDetail = casesDetails.get(i);
            try {
                log.info("Access migration - migrating access for case {}. ({}/{})",
                    casesDetail.getId(), i + 1, casesDetailsSize);

                reShareCases(casesDetail);
            } catch (Exception e) {
                log.error("Access migration failed for case {}", casesDetail.getId(), e);
            }
        }
```
and remove the need for an additional counter

Reply:
does not really matter, wanted to make minimal changes to the tested version. But yeah, changed